app/src/main/java/com/example/pygmyhippo/common/NotificationCenter.java

```diff
@@ -0,0 +1,200 @@
+    private EventDB dbHandler;
+    private String accountID;
+    private int nextNotificationId;
+
+    public NotificationCenter(Activity activity, String accountID) {
+        this.activity = activity;
+        this.accountID = accountID;
+        this.context = activity.getApplicationContext();
+        this.dbHandler = new EventDB();
+        this.nextNotificationId = 0;
+    }
+
+    /**
+     * Initialization function. Asks for permissions and creates notification channels if necessary.
+     */
+    public void init() {
+        getPermission();
+        createNotificationChannel();
+    }
+
+    /**
+     * Asks for post notification permissions.
+     *
+     * This is only necessary in SDK versions > 33.
+     */
+    public void getPermission() {
+        try {
+            Log.d("NotificationCenter", "Init");
+            if (Build.VERSION.SDK_INT > Build.VERSION_CODES.TIRAMISU) {
+                ActivityCompat.requestPermissions(activity,
+                        new String[]{Manifest.permission.POST_NOTIFICATIONS},
+                        PERMISSION_REQUEST_CODE);
+            }
+        }catch (Exception e){
+            Log.d("NotificationCenter", "Exception when requesting notification permissions");
+        }
+    }
+
+    /**
+     * Creates a notification channel for posting notifications.
+     *
+     * This is necessary to post notifications in SDK versions >= 26
+     */
+    private void createNotificationChannel() {
+        // Create the NotificationChannel, but only on API 26+ because
+        // the NotificationChannel class is not in the Support Library.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            Log.d("MainActivity", "Creating notification channel");
+
+            CharSequence name = context.getString(R.string.notification_channel_name);
+            String description = context.getString(R.string.notification_channel_description);
+            int importance = NotificationManager.IMPORTANCE_DEFAULT;
+            String channel_id = context.getString(R.string.notification_channel_ID);
+
+            NotificationChannel channel = new NotificationChannel(channel_id, name, importance);
+            channel.setDescription(description);
+            NotificationManager notificationManager = context.getSystemService(NotificationManager.class);
+            notificationManager.createNotificationChannel(channel);
+        }
+    }
+
+    /**
+     * Posts a notification.
+     * @param title - Title of notification.
+     * @param content - Body of the notification.
+     * @return notification ID to be used elsewhere.
+     */
+    private int postNotification(String title, String content) {
+        String channel_id = context.getString(R.string.notification_channel_ID);
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(activity, channel_id);
+        builder.setContentTitle(title)
+                .setSmallIcon(R.drawable.hippoparty)
```

Comment:
does it make sense to have the notification icon be the poster from the default event? maybe something like moo_deng.jpg is better suited here

Reply:
maybe it could be the event icon

Reply:
The current icon is a placeholder. Will replace with the app icon if we ever get one